galgebra/metric.py

```diff
@@ -690,37 +691,38 @@ def __init__(
                     raise ValueError('For metric derived from vector field ' +
                                      ' coordinates must be defined.')
                 else:  # Vector manifold defined by vector field
-                    dX = []
-                    for coord in coords:  # Get basis vectors by differentiating vector field
-                        dX.append([diff(x, coord) for x in X])
-                    g_tmp = []
-                    for dx1 in dX:
-                        g_row = []
-                        for dx2 in dX:
-                            dx1_dot_dx2 = trigsimp(Metric.dot_orthogonal(dx1, dx2, g))
-                            g_row.append(dx1_dot_dx2)
-                        g_tmp.append(g_row)
-                    self.g = Matrix(g_tmp)
+                    # Get basis vectors by differentiating vector field
+                    dX = [
+                        [diff(x, coord) for x in X]
+                        for coord in coords
+                    ]
+                    self.g = Matrix([
+                        [
+                            trigsimp(Metric.dot_orthogonal(dx1, dx2, g))
+                            for dx2 in dX
+                        ]
+                        for dx1 in dX
+                    ])
                     if self.debug:
                         printer.oprint('X_{i}', X, 'D_{i}X_{j}', dX)
 
         else:  # metric is symbolic or list of lists of functions of coordinates
             if isinstance(g, str):  # metric elements are symbols or constants
                 if g == 'g':  # general symbolic metric tensor (g_ij functions of position)
-                    g_lst = []
-                    g_inv_lst = []
-                    for coord in self.coords:
-                        i1 = str(coord)
-                        tmp = []
-                        tmp_inv = []
-                        for coord2 in self.coords:
-                            i2 = str(coord2)
-                            tmp.append(Function('g_'+i1+'_'+i2)(*self.coords))
-                            tmp_inv.append(Function('g__'+i1+'__'+i2)(*self.coords))
-                        g_lst.append(tmp)
-                        g_inv_lst.append(tmp_inv)
-                    self.g = Matrix(g_lst)
-                    self.g_inv = Matrix(g_inv_lst)
+                    self.g = Matrix([
+                        [
+                            Function('g_{}_{}'.format(coord, coord2))(*self.coords)
+                            for coord2 in self.coords
+                        ]
+                        for coord in self.coords
+                    ])
+                    self.g_inv = Matrix([
+                        [
+                            Function('g__{}__{}'.format(coord, coord2))(*self.coords)
+                            for coord2 in self.coords
+                        ]
+                        for coord in self.coords
+                    ])
```

Comment:
These lines aren't covered

Reply:
`g = Ga('e*1|2', g='g', coords=sympy.symbols('x y'))` exercises it and doesn't crash, although it seems `g.g_inv` gets overwritten anyway.